Piotr/optimize search selections with a limit

osiewicz opened the pull request:
/cc @nathansobo @maxbrunsfeld 

Release Notes:
- Fixed scrollbar selections causing noticeable slowdowns with large quantities of selections.

osiewicz committed:
Baseline: Improve selection rendering for large quantities from 270ms to 90ms

osiewicz committed:
Do not query start of range if it's end is the same as the previous hunk's

osiewicz committed:
Add upper bound limit. Remove dbg! statements

osiewicz committed:
Remove redundant argument

osiewicz committed:
Use display maps

maxbrunsfeld reviewed (commented):
Thoughts:

* I think the name `selected_rows` makes it sound like these ranges are selections, but actually they're background highlights. What do you think about naming the method something like `background_highlight_row_ranges`, since this method is converting the background highlighted anchor ranges into  row ranges?
* If there are more search results than the limit of 50k, I think we may want to not show *any* results in the scrollbar, instead of showing a subset of the search results in the scrollbar. What do you think?
* Could you add a test for this method? I think you could manually create highlights with `Editor::highlight_background`.

osiewicz committed:
Address review feedback.

Rename selected_rows to background_highlight_row_ranges.
Do not return any ranges if there are more than 50k results

osiewicz committed:
Do not highlgiht selections at all over the threshold
